### PR TITLE
Database (JSON) improvements

### PR DESCRIPTION
**Pull request by sfayer**
Allow database objects to be passed to jsonify.
Add a new (optional) attribute to database objects '__json_fields__' to limit which fields get converted to JSON when .json() or jsonify() is called.
Add test cases for all of the framework.Database classes.

### CODE REVIEW

File: src/pdm/framework/Database.py [hunk 1]
```diff
@@ -31,8 +31,12 @@ def default(self, obj):
...
             return obj.isoformat()
         if isinstance(obj, JSONMixin):
-            return {column.name: getattr(obj, column.name)
-                    for column in obj.__table__.columns}
+            if hasattr(obj, '__json_fields__'):
+                return {attr_name: getattr(obj, attr_name)
+                        for attr_name in obj.__json_fields__}
+            else:
+                return {column.name: getattr(obj, column.name)
+                        for column in obj.__table__.columns}
```
Comment (alexanderrichards) [inline src/pdm/framework/Database.py]: The reason I didn't approve this straight away is I feel there should be a nice way to write this. Looking again the difficulty Congress from needing to use column.name Vs simply attr_name. I wrote it in a simpler way only to then realise this problem.

Comment (sfayer) [reply src/pdm/framework/Database.py]: Yes, I thought you were probably thinking about it... There isn't any rush as Janusz has a workaround now, so we can leave it for a few days and see if anyone thinks of any neater ideas.

Comment (sfayer) [reply src/pdm/framework/Database.py]: What if we inverted the list to be something like \_\_excluded_fields\_\_? Then we could just create a list of columns using a list comprehension to filter it (and then used the dictionary comprehension to create the return value as before). If we default \_\_excluded_fields\_\_ to an empty list in the JSONMixin class then we can avoid the edge case of classes that don't specify it.

File: src/pdm/framework/Database.py [hunk 2]
```diff
@@ -31,12 +31,9 @@ def default(self, obj):
...
-            if hasattr(obj, '__json_fields__'):
-                return {attr_name: getattr(obj, attr_name)
-                        for attr_name in obj.__json_fields__}
-            else:
-                return {column.name: getattr(obj, column.name)
-                        for column in obj.__table__.columns}
+            cols = [column.name for column in obj.__table__.columns
+                                if column.name not in obj.__excluded_fields__]
+            return {column: getattr(obj, column) for column in cols}
         return super(JSONTableEncoder, self).default(obj)
```
Comment (alexanderrichards) [inline src/pdm/framework/Database.py]: Yes this is the way I was going as well. I think this is better

**Pull request merged**